packages/flutter/lib/src/material/time.dart

```diff
@@ -134,6 +135,43 @@ class TimeOfDay {
   }
 }
 
+/// A [RestorableValue] that knows how to save and restore [TimeOfDay].
+///
+/// {@macro flutter.widgets.RestorableNum}.
+class RestorableTimeOfDay extends RestorableValue<TimeOfDay> {
+  /// Creates a [RestorableTimeOfDay].
+  ///
+  /// {@macro flutter.widgets.RestorableNum.constructor}
+  RestorableTimeOfDay(TimeOfDay defaultValue) : _defaultValue = defaultValue;
+
+  final TimeOfDay _defaultValue;
+
+  @override
+  TimeOfDay createDefaultValue() => _defaultValue;
+
+  @override
+  void didUpdateValue(TimeOfDay? oldValue) {
+    assert(debugIsSerializableForRestoration(value.hour));
+    assert(debugIsSerializableForRestoration(value.minute));
+    notifyListeners();
+  }
+
+  @override
+  TimeOfDay fromPrimitives(Object? data) {
+    final Map<String, int> timeData = Map<String, int>.from(data! as Map<Object?, Object?>);
+    return TimeOfDay(
+      minute: timeData['minute']!,
+      hour: timeData['hour']!,
+    );
+  }
+
+  @override
+  Object? toPrimitives() => <String, int>{
+    'minute': value.minute,
+    'hour': value.hour,
+  };
```

Comment:
We could eliminate the keys and just store this as:

```suggestion
  Object? toPrimitives() => <int>[value.minute, value.hour];
```

... with a corresponding change in fromPrimitives.